spec/opentsdb_consumer_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'spec_helper'
+
+RSpec.describe OpenTSDBConsumer do
+  let(:host)         { 'myhost.com' }
+  let(:port)         { 4242 }
+  let(:metric_names) { %w(metric1 metric2) }
+
+  describe '.fetch' do
+    subject do
+      described_class.fetch(
+        host: host, port: port,
+        metric_names: metric_names,
+        metric_options: { aggregator: 'cubic', start: 'now', unknown_key: 'not here' }
+      )
+    end
+
+    before(:each) do
+      allow(OpenTSDBConsumer::Client).to receive(:new)
+      allow(OpenTSDBConsumer::Metric).to receive(:new)
+
+      query_double = instance_double(OpenTSDBConsumer::Query)
```

Comment:
Is it really necessary to use doubles everywhere? We should try hard to just use them where we have no other choice (when calling something outside for example, which the Client class does)